### PR TITLE
fix(core): strip ansi before checking logfilters

### LINKED ISSUE #2204: [Bug?] `logFilters` doesn't work with resolution warnings
The resolution step produces this warning in our case:

> ➤ YN0060: │ @company/my-package@workspace:my-package provides graphql (pc69a0) with version 14.7.0, which doesn't satisfy what @graphql-codegen/typescript-operations and some of its descendants request

I tried silencing it via `logFilters` but no success:

```
logFilters:
  - text: "satisfy what"
    level: discard
```

(I tried several strings like `"provides graphql"` or `"with version 14.7.0"`, no luck either.)

Should `logFilters` be supported for resolution warnings? I'm not sure if this is a bug or expected behavior.

### PR DESCRIPTION
**Pull request by tim-stasse**
**What's the problem this PR addresses?**
Ansi codes (from chalk) weren't being removed when comparing log text with the configured log filter text, as described in [chalk #62](https://github.com/chalk/chalk/issues/62#issuecomment-96895546)

Closes #2204

...

**How did you fix it?**
Used `strip-ansi` to remove ansi codes ([chalk #62](https://github.com/chalk/chalk/issues/62#issuecomment-417170880))

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: .yarn/versions/2e696b36.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+releases:
+  "@yarnpkg/builder": patch
```
Comment (merceyz) [inline .yarn/versions/2e696b36.yml:2]: Only need a patch release on core and cli, the rest can be declined

**Comment by merceyz** [comment]
Could you add a test to https://github.com/yarnpkg/berry/blob/d96f75eb53445ec5297fb9885e186eb437ed5a15/packages/acceptance-tests/pkg-tests-specs/sources/features/logFilter.test.js please

File: .yarn/versions/2e696b36.yml (hunk 1 above)
Comment (tim-stasse) [reply .yarn/versions/2e696b36.yml:2]: Ah cool, gotcha! Thanks 👍

**Comment by tim-stasse** [comment]
Can do 👍

**Comment by tim-stasse** [comment]
Confirmed the added test fails on master

**Comment by arcanis** [comment]
Perfect, thanks!

**Pull request merged**